pkg/grpc/api.go

```diff
@@ -565,17 +565,18 @@ func (a *api) GetBulkSecret(ctx context.Context, in *runtimev1pb.GetBulkSecretRe
 		return &runtimev1pb.GetBulkSecretResponse{}, err
 	}
 
-	for key := range getResponse.Data {
-		if !a.isSecretAllowed(in.StoreName, key) {
-			err := status.Errorf(codes.PermissionDenied, messages.ErrPermissionDenied, key, in.StoreName)
-			apiServerLogger.Debug(err)
-			return &runtimev1pb.GetBulkSecretResponse{}, err
+	filteredSecrets := map[string]string{}
+	for key, v := range getResponse.Data {
+		if a.isSecretAllowed(secretStoreName, key) {
+			filteredSecrets[key] = v
+		} else {
+			apiServerLogger.Warnf(messages.ErrPermissionDenied, key, in.StoreName)
```

Comment:
I don't think we should log a warning here. Only returning the allowed secrets is the expected result.

Reply:
I agree with @msfussell's https://github.com/dapr/dapr/issues/2600#issuecomment-754197385 on logging the warnings. The warnings may help to fine tune the solution and create a store with just the required secrets for the application and avoid returning them just to be discarded by the filter.

Reply:
I'm a bit torn on this one. Yes, it will definitely help with diagnostics and fine-tuning so that's good. On the other hand, if you use any restrictions in your application, you now end up with always having "false-positive" warnings in your logs that you can ignore. 

@yaron2 What's your take on this?

Reply:
I'd offer a middleground approach: I think we should log, but those should probably be `debug` level logs.

Reply:
done!